Increase our default/minimum dust limit and decrease our max

TheBlueMatt opened the pull request:
CC @ariard can you take a look at the test I dropped here - its not documented how you came up with the constants used so its rather difficult to fix the test. Would be nice to replace all the constants with calculation based on the dust limits set.

ariard commented on lightning/src/ln/channel.rs:
```diff
@@ -571,21 +571,20 @@ pub const COMMITMENT_TX_WEIGHT_PER_HTLC: u64 = 172;
+/// P2SH outputs).
+/// We also use this as the maximum counterparty `dust_limit_satoshis` allowed, given many nodes
+/// use this value for their dust limit today.
+pub const MAX_STD_OUTPUT_DUST_LIMIT_SATOSHIS: u64 = 546;
```
nit: s/nodes/implementations/g. IIRC `dust_limit_satoshis` isn't configurable on many implementations but we should dissociate node operators policy from implemenation setting IMO. Even if we can assume A matches B, 99%.

ariard commented on lightning/src/ln/channel.rs:
```diff
@@ -571,21 +571,20 @@ pub const COMMITMENT_TX_WEIGHT_PER_HTLC: u64 = 172;
+
+/// The dust limit is used for both the commitment transaction outputs as well as the closing
+/// transactions. For cooperative closing transactions, we require segwit outputs, though accept
+/// *any* segwit scripts, which are allowed to be up to 42 bytes in thelgth.
```
nit: s/thelgth/length/g

ariard commented on lightning/src/ln/channel.rs:
```diff
@@ -3665,6 +3664,12 @@ impl<Signer: Sign> Channel<Signer> {
 		};
 
+		for outp in closing_tx.output.iter() {
+			if !outp.script_pubkey.is_witness_program() && outp.value < MAX_STD_OUTPUT_DUST_LIMIT_SATOSHIS {
```
Wait, the closing requirement implemented by this change is "Require cooperative close payout scripts to be Segwit" ? If so why also checking `MAX_STD_OUTPUT_DUST_LIMIT_SATOSHIS`, we want to reject non-segwit output no matter their output values ?

TheBlueMatt replied:
The new requirement is "our users have to use segwit scripts, but the counterparty can use something else, but if they do we may have to force close". I tweaked the commit message text to be more clear.

ariard reviewed (commented):
Concept ACK,

I can take #1065 on top of #1054 and fix the test there. Yeah values are not documented because...though the easy fix should be just to sub  (660 - 543) to the loading HTLC amounts. I think.

TheBlueMatt commented:
Rebased and squashed the fixup commits. I think we should go ahead and take this and just open an issue for the broken test to re-add it later - without this we're going to be unable to open any channels with other nodes very soon.

valentinewallace reviewed (commented):
One suggestion, otherwise looks good

TheBlueMatt commented on lightning/src/ln/channel.rs:
```diff
@@ -555,15 +555,12 @@ pub const ANCHOR_OUTPUT_VALUE_SATOSHI: u64 = 330;
+/// transaction non-standard and thus refuses to relay it.
+/// We also use this as the maximum counterparty `dust_limit_satoshis` allowed, given many
+/// implementations use this value for their dust limit today.
+pub const MAX_STD_OUTPUT_DUST_LIMIT_SATOSHIS: u64 = 546;
```
Did something similar which I think captured your intent, lmk.

valentinewallace replied:
Oops, I meant in my original diff to rename the new const back to `MAX_DUST_LIMIT_SATOSHIS` (since it seems like an implementation detail what the actual value is)

TheBlueMatt replied:
Ah, ok, i see your point now. reverted the fixup, take a look now.

valentinewallace commented on lightning/src/ln/channel.rs:
```diff
@@ -555,21 +554,24 @@ pub const ANCHOR_OUTPUT_VALUE_SATOSHI: u64 = 330;
+pub const MAX_STD_OUTPUT_DUST_LIMIT_SATOSHIS: u64 = 546;
+
+/// The maximum channel dust limit we will accept from our counterparty.
+pub const MAX_CHAN_DUST_LIMIT_SATOSHIS: u64 = MAX_STD_OUTPUT_DUST_LIMIT_SATOSHIS;
```
For conciseness, why not `MAX_DUST_LIMIT_SATOSHIS`?

valentinewallace replied:
Isn't `MIN_DUST_LIMIT_SATOSHIS` also a channel dust limit? Mainly the inconsistency between them I find confusing

ariard replied:
Yes I agree `MIN_DUST_LIMIT_SATOSHIS` could be `MIN_CHAN_DUST_LIMIT_SATOSHIS` as it's applied at the minimal channel's `dust_limit_satoshis` accepted by our implementation.

TheBlueMatt replied:
Pushed a commit to rename that, too :)

ariard reviewed (commented):
Code Review ACK 88c0619b

TheBlueMatt commented:
Squashed with no diff. Diff since ariard's ack is a trivial constant rename. Will land after CI.

```
$ git diff-tree -U0 88c0619 2da0d6c0
diff --git a/lightning/src/ln/channel.rs b/lightning/src/ln/channel.rs
index 3faaafe8..f412bf2f 100644
--- a/lightning/src/ln/channel.rs
+++ b/lightning/src/ln/channel.rs
@@ -574 +574 @@ pub const MAX_CHAN_DUST_LIMIT_SATOSHIS: u64 = MAX_STD_OUTPUT_DUST_LIMIT_SATOSHIS
-pub const MIN_DUST_LIMIT_SATOSHIS: u64 = 354;
+pub const MIN_CHAN_DUST_LIMIT_SATOSHIS: u64 = 354;
@@ -641 +641 @@ impl<Signer: Sign> Channel<Signer> {
-		if holder_selected_channel_reserve_satoshis < MIN_DUST_LIMIT_SATOSHIS {
+		if holder_selected_channel_reserve_satoshis < MIN_CHAN_DUST_LIMIT_SATOSHIS {
@@ -712 +712 @@ impl<Signer: Sign> Channel<Signer> {
-			holder_dust_limit_satoshis: MIN_DUST_LIMIT_SATOSHIS,
+			holder_dust_limit_satoshis: MIN_CHAN_DUST_LIMIT_SATOSHIS,
@@ -846,2 +846,2 @@ impl<Signer: Sign> Channel<Signer> {
-		if msg.dust_limit_satoshis < MIN_DUST_LIMIT_SATOSHIS {
-			return Err(ChannelError::Close(format!("dust_limit_satoshis ({}) is less than the implementation limit ({})", msg.dust_limit_satoshis, MIN_DUST_LIMIT_SATOSHIS)));
+		if msg.dust_limit_satoshis < MIN_CHAN_DUST_LIMIT_SATOSHIS {
+			return Err(ChannelError::Close(format!("dust_limit_satoshis ({}) is less than the implementation limit ({})", msg.dust_limit_satoshis, MIN_CHAN_DUST_LIMIT_SATOSHIS)));
@@ -867,2 +867,2 @@ impl<Signer: Sign> Channel<Signer> {
-		if holder_selected_channel_reserve_satoshis < MIN_DUST_LIMIT_SATOSHIS {
-			return Err(ChannelError::Close(format!("Suitable channel reserve not found. remote_channel_reserve was ({}). dust_limit_satoshis is ({}).", holder_selected_channel_reserve_satoshis, MIN_DUST_LIMIT_SATOSHIS)));
+		if holder_selected_channel_reserve_satoshis < MIN_CHAN_DUST_LIMIT_SATOSHIS {
+			return Err(ChannelError::Close(format!("Suitable channel reserve not found. remote_channel_reserve was ({}). dust_limit_satoshis is ({}).", holder_selected_channel_reserve_satoshis, MIN_CHAN_DUST_LIMIT_SATOSHIS)));
@@ -870,2 +870,2 @@ impl<Signer: Sign> Channel<Signer> {
-		if msg.channel_reserve_satoshis < MIN_DUST_LIMIT_SATOSHIS {
-			return Err(ChannelError::Close(format!("channel_reserve_satoshis ({}) is smaller than our dust limit ({})", msg.channel_reserve_satoshis, MIN_DUST_LIMIT_SATOSHIS)));
+		if msg.channel_reserve_satoshis < MIN_CHAN_DUST_LIMIT_SATOSHIS {
+			return Err(ChannelError::Close(format!("channel_reserve_satoshis ({}) is smaller than our dust limit ({})", msg.channel_reserve_satoshis, MIN_CHAN_DUST_LIMIT_SATOSHIS)));
@@ -976 +976 @@ impl<Signer: Sign> Channel<Signer> {
-			holder_dust_limit_satoshis: MIN_DUST_LIMIT_SATOSHIS,
+			holder_dust_limit_satoshis: MIN_CHAN_DUST_LIMIT_SATOSHIS,
@@ -1620,2 +1620,2 @@ impl<Signer: Sign> Channel<Signer> {
-		if msg.dust_limit_satoshis < MIN_DUST_LIMIT_SATOSHIS {
-			return Err(ChannelError::Close(format!("dust_limit_satoshis ({}) is less than the implementation limit ({})", msg.dust_limit_satoshis, MIN_DUST_LIMIT_SATOSHIS)));
+		if msg.dust_limit_satoshis < MIN_CHAN_DUST_LIMIT_SATOSHIS {
+			return Err(ChannelError::Close(format!("dust_limit_satoshis ({}) is less than the implementation limit ({})", msg.dust_limit_satoshis, MIN_CHAN_DUST_LIMIT_SATOSHIS)));
diff --git a/lightning/src/ln/functional_tests.rs b/lightning/src/ln/functional_tests.rs
index 0991434c..e6529415 100644
--- a/lightning/src/ln/functional_tests.rs
+++ b/lightning/src/ln/functional_tests.rs
@@ -1483 +1483 @@ fn test_chan_reserve_dust_inbound_htlcs_outbound_chan() {
-	let dust_amt = crate::ln::channel::MIN_DUST_LIMIT_SATOSHIS * 1000
+	let dust_amt = crate::ln::channel::MIN_CHAN_DUST_LIMIT_SATOSHIS * 1000
$
```
